packages/hooks/src/__tests__/useSettings.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useQuery, useMutation, useInvalidateQuery } from '@deriv/api';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import useSettings from '../useSettings';
+
+jest.mock('@deriv/api', () => ({
+    ...jest.requireActual('@deriv/api'),
+    useQuery: jest.fn(),
+    useMutation: jest.fn(),
+    useInvalidateQuery: jest.fn(),
+}));
+
+describe('useSettings', () => {
+    const mockInvalidateQuery = jest.fn();
+    const mockMutate = jest.fn();
+    const mockStoreData = mockStore({
+        client: { is_authorize: true },
+    });
+
+    beforeEach(() => {
+        (useInvalidateQuery as jest.Mock).mockReturnValue(mockInvalidateQuery);
+        (useMutation as jest.Mock).mockReturnValue({
+            mutate: mockMutate,
+            data: null,
+            isLoading: false,
+            error: null,
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should return initial state correctly', () => {
+        (useQuery as jest.Mock).mockReturnValue({
+            data: null,
+            isLoading: false,
+            error: null,
+        });
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
```

Comment:
let's move this code to separate function and reuse it, rather than create wrapped in every test